src/System.Diagnostics.Process/src/System/Collections/Specialized/DictionaryWrapper.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Collections.Specialized
+{
+    [Serializable]
+    internal sealed class DictionaryWrapper : IDictionary<string, string>, IDictionary
+    {
+        private readonly Dictionary<string, string> _contents;
+
+        public DictionaryWrapper(Dictionary<string, string> contents)
+        {
+            _contents = contents;
+        }
+
+        public string this[string key]
+        {
+            get => _contents[key];
+            set => _contents[key] = value;
+        }
+
+        public object this[object key]
+        {
+            get => this[(string)key];
+            set => this[(string)key] = (string)value;
+        }
+
+        public ICollection<string> Keys => _contents.Keys;
+        public ICollection<string> Values => _contents.Values;
+
+        ICollection IDictionary.Keys => _contents.Keys;
+        ICollection IDictionary.Values => _contents.Values;
+
+        public int Count => _contents.Count;
+
+        public bool IsReadOnly => ((IDictionary)_contents).IsReadOnly;
+        public bool IsSynchronized => ((IDictionary)_contents).IsSynchronized;
+        public bool IsFixedSize => ((IDictionary)_contents).IsFixedSize;
+        public object SyncRoot => ((IDictionary)_contents).SyncRoot;
+
+        public void Add(string key, string value)
+        {
+            // If the key already exists, overwrite it.
+            if (ContainsKey(key))
```

Comment:
No need to check for presence of key, can just do this[key] = value. this indexer set doesn't throw for duplicates, rewrites the value.